lib/Transforms/Scalar/DxilLoopUnroll.cpp

```diff
@@ -683,22 +682,32 @@ static void RecursivelyRemoveLoopFromQueue(LPPassManager &LPM, Loop *L) {
 
 bool DxilLoopUnroll::runOnLoop(Loop *L, LPPassManager &LPM) {
 
+  DebugLoc LoopLoc = L->getStartLoc(); // Debug location for the start of the loop.
+  Function *F = L->getHeader()->getParent();
+
   bool HasExplicitLoopCount = false;
-  unsigned UnrollCount = 0;
+  int ExplicitUnrollCountSigned = 0;
 
   // If the loop is not marked as [unroll], don't do anything.
-  if (IsMarkedUnrollCount(L, &UnrollCount)) {
+  if (IsMarkedUnrollCount(L, &ExplicitUnrollCountSigned)) {
     HasExplicitLoopCount = true;
   }
   else if (!IsMarkedFullUnroll(L)) {
     return false;
   }
 
+  unsigned ExplicitUnrollCount = 0;
+  if (HasExplicitLoopCount) {
+    if (ExplicitUnrollCountSigned < 1) {
+      FailLoopUnroll(false, F->getContext(), LoopLoc, "Could not unroll loop. Invalid unroll count.");
```

Comment:
Could we fail during semantic analysis instead?

Reply:
That would be ideal. I'll make it an error instead of a warning that in a different change.